query/query.go

```diff
@@ -991,12 +993,12 @@ func calculateFirstN(sg *SubGraph) int32 {
 	}
 
 	if len(sg.Filters) == 0 && len(sg.Params.Order) == 0 && !shouldExclude {
-		// Offset also added because, we need n results to trim the offset.
 		if sg.Params.Count != 0 {
-			count = sg.Params.Count + sg.Params.Offset
+			return int32(sg.Params.Count), int32(sg.Params.Offset)
 		}
 	}
-	return int32(count)
+	// make offset = 0, if there is need to fetch all the results.
+	return int32(count), 0
```

Comment:
This should probably be:
        return int32(count), int32(sg.Params.Offset)
There is a case where one might want all the results after skipping the first few (denoted by the Offset).

Reply:
@akon-dey Actually no, I think that in this case worker/task.go needs `pb.query.Offset` to in fact be 0 in order to function correctly.